File: src/main/java/com/silverboxsoft/classes/DynamoDbCondition.java
package com.silverboxsoft.classes;

import lombok.Data;

@Data
public class DynamoDbCondition {
	private String columnName;
	private DynamoDbConditionType conditionType;
	private String value;

	public String getConditionExpression() {
		return getAlias().concat(conditionType.getCondStr()).concat(":").concat(columnName);
	}

	public String getAlias() {
		return "#".concat(columnName);
	}
}

File: src/main/java/com/silverboxsoft/dynamodbtool/classes/DynamoDbCondition.kt
package com.silverboxsoft.dynamodbtool.classes

import lombok.Data

@Data
class DynamoDbCondition(columnName: String, conditionType: DynamoDbConditionType, value: String) {
    private val columnName: String = columnName
    private val conditionType: DynamoDbConditionType = conditionType
    private val value: String = value
    val conditionExpression: String
        get() = alias + conditionType.getCondStr() + ":" + columnName
    val alias: String
        get() = "#$columnName"
}